src/Components/Search/SearchInputContainer.tsx

```diff
@@ -0,0 +1,44 @@
+import { Box, color } from "@artsy/palette"
+import Input from "Components/Input"
+import React from "react"
+import styled from "styled-components"
+
+const SearchButton = styled.button`
+  position: absolute;
+  right: 0;
+  top: 50%;
+  border: none;
+  margin-top: -14px;
+  width: 24px;
+  height: 24px;
+  margin-right: 10px;
+  background: none;
+  padding: 0;
+
+  :focus {
+    background: ${color("purple100")};
+    border-radius: 50%;
+
+    svg > path {
+      fill: ${color("white100")};
+    }
+  }
+`
+
+export const SearchInputContainer: React.ExoticComponent<any> = React.forwardRef((props, ref) => {
```

Comment:
oo so what's this do? This is the second time I've seen this used to avoid that `stateless ref` message, which I encounter often, but am still unclear what's going on. Looks exotic tho! 😎

Reply:
I don't fully understand the `ExoticComponent` bit, but they talk about it here:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/aee81ea6a246aea9f3ec957ad968d8500e18de22/types/react/index.d.ts#L286-L296

I guess I need it because `forwardRef` doesn't return a simple component. And the reason I need `forwardRef` is so that I can bubble the ref up to Autosuggest.